packages/rabbitmq/src/amqp/connection.ts

```diff
@@ -405,20 +414,28 @@ export class AmqpConnection {
       msgOptions,
       channel,
     });
+
+    return consumerTag;
```

Comment:
What i meant is that the return type is changed to an object that has a `consumerTag` property, you can create a new interface for the return type so that it become easily scalable

Reply:
Right - I exposed it as an interface on any `public` methods when we're returning the tag to consumers, but left it as a single return value on any internal functions. Would you prefer it be wrapped in an interface for the `private` functions as well?

Reply:
Yes i saw that and it looks good for the callback but i also believe it would look okay in the return type of the function itself, if the previous type were `void` then we are not going to break anything since the current version does not expose anything. My point of view here is merely scalability, if people start consuming this right away and in the future, we want to explore more properties out of the channel then it would break right away every consumer

Reply:
Are you suggesting that the return type of `createSubscriber()` be updated from `Promise<void>` to `Promise<SubscriptionResult>`, or that the internal `setupSubscriberChannel()` return type be updated from `Promise<ConsumerTag>` to `Promise<SubscriptionResult>`?

Reply:
I believe only the public API because the internal methods will not use it (but. yeah something like SubscriptionResult)

Reply:
Got it. The issue is we don't have the ConsumerTag at that point - our public method body looks like:
```
return this.selectManagedChannel(
      msgOptions?.queueOptions?.channel
    ).addSetup(async (channel) => {
      const consumerTag = await this.setupSubscriberChannel<T>(...);
    });
)
```
And the `consumerTag` comes _back to us_ inside a callback - not as a return value. Bit of context [here](https://github.com/golevelup/nestjs/issues/415#issuecomment-1191579258), but if there's a workaround or something I'm missing, I'd definitely also prefer to return `Promise<SubscriptionResult>` instead of `Promise<void>` with a callback.

Reply:
I see. I'll leave this up to your judgment, I think people will probably come up with more ideas when consuming this so

Reply:
@underfisk I'm trying the `return Promise(...)` route, would you mind kicking off the CI? 🙏